core/app/services/workarea/direct_upload.rb

```diff
@@ -2,15 +2,14 @@ module Workarea
   class DirectUpload
     class InvalidTypeError < RuntimeError; end
 
-    def self.ensure_cors!
-      url = "http#{Rails.application.config.force_ssl ? 's' : ''}://"
-      url += Workarea.config.host
+    def self.ensure_cors!(url)
+      uri = URI.parse(url)
```

Comment:
We can just get rid of this right? No need for just the host below, might as well use the full URL as the ID in the config.